pkg/apis/openebs.io/scp/v1alpha1/storage-volume.go

```diff
@@ -39,3 +44,266 @@ type StorageVolumeList struct {
 
 	Items []StorageVolume `json:"items"`
 }
+
+// StorageVolumeSpec defines the desired spec of StorageVolume
+// +k8s:openapi-gen=true
+type StorageVolumeSpec struct {
+	// Capacity of the volume
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	Capacity resource.Quantity `json:"capacity"`
+
+	// ProtocolConfiguration represents protocol specific configurations for NVME
+	// +nullable
+	ProtocolConfiguration ProtocolConfiguration `json:"protocolConfiguration,omitempty"`
+
+	// VolumeType stores configuration that is passed through storage class
+	VolumeType VolumeType `json:"volumeType"`
+
+	// Affinity is a group of affinity scheduling rules
+	// +nullable
+	Affinity Affinity `json:"affinity,omitempty"`
+
+	// StorageCohortID represents the cohort in which volume has to be provisioned
+	StorageCohortName string `json:"storageCohortName"`
+
+	// StoragePoolID represents the pool in which volume has to be created
+	StoragePoolName string `json:"storagePoolName"`
+}
+
+// ProtocolConfiguration represents protocol specific configurations for NVME
+type ProtocolConfiguration struct {
+	// AllowedHosts represents the allowed hosts
+	AllowedHosts []string `json:"allowedHosts,omitempty"`
+}
+
+// VolumeType stores configuration of volumes that is passed through storage class
+type VolumeType struct {
```

Comment:
Wondering if we need a separate volumeType if it is the same as capabilities, cc @shovanmaity ?

Reply:
:+1: 
we can name the file as common.go and put all common stuff.

Reply:
Ok, what I actually meant in this case, is that we can directly use capabilities itself instead of volumetype since volumetype is also having only one object i.e. capabilities.

Reply:
Ok yes we should do that.